name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/PDFParser.java

```diff
@@ -26,19 +26,24 @@
     /* package */ static class DocumentType
     {
         private List<Pattern> mustInclude = new ArrayList<>();
+        private List<Pattern> mustNotInclude = new ArrayList<>();
 
         private List<Block> blocks = new ArrayList<>();
         private DocumentContext context = new DocumentContext();
         private BiConsumer<DocumentContext, String[]> contextProvider;
 
-        public DocumentType(List<Pattern> mustInclude)
+        public DocumentType(String mustInclude)
         {
-            this.mustInclude.addAll(mustInclude);
+            this.mustInclude.add(Pattern.compile(mustInclude));
         }
 
-        public DocumentType(String mustInclude)
+        public DocumentType(String mustInclude, String mustNotInclude)
         {
-            this(mustInclude, null);
+            this.mustInclude.add(Pattern.compile(mustInclude));
+            if (mustNotInclude != null)
+            {
+                this.mustNotInclude.add(Pattern.compile(mustNotInclude));
+            }
```

Comment:
I propose to remove this null check.
If one does not have an exclude pattern ("mustNotInclude") then do not use this constructor.

Reply:
done... i remove this null check